effectful-th/src/Effectful/TH.hs

```diff
@@ -0,0 +1,238 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE TemplateHaskell #-}
```

Comment:
Can you replace this with [TemplateHaskellQuotes](https://mpickering.github.io/ghc-docs/build-html/users_guide/glasgow_exts.html#extension-TemplateHaskellQuotes)?

Reply:
Unfortunately that is not possible. Changing that to `TemplateHaskellQuotes` results in:
```
src/Effectful/TH.hs:156:40: error:
    Operator applied to too few arguments: :>
    |
156 |       effectConstraint = [t| $(effect) :> $(varT es) |]
    |
```

Reply:
This requires getting rid of splices, but it should be possible as you can construct the AST manually. FWIW `optics-th` uses TemplateHaskellQuotes only. This matter for cross-compilation.